plot_wavefield uses `specfem::program::abort` but `periodic_tasks` has no library for it

`dim3/plot_wavefield.cpp` has an `abort` call here:

https://github.com/PrincetonUniversity/SPECFEMPP/blob/0f23f8f60171921441f0041d9d99c859b9f04859/core/specfem/periodic_tasks/plot_wavefield/dim3/plot_wavefield.cpp#L426-L438

 but no library with the `abort` function is linked (e.g. `specfem_program`). Note that this is behind an `else` clause on `#ifdef NO_VTK`.

For example, on my system:

```bash
/usr/bin/ld: archive/libperiodic_tasks.a(plot_wavefield.cpp.o): in function `void specfem::periodic_tasks::plot_wavefield<(specfem::dimension::type)1>::initialize<(specfem::display::format)3>(vtkSmartPointer<vtkFloatArray>&)':
/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2/core/specfem/periodic_tasks/plot_wavefield/dim3/plot_wavefield.cpp:435:(.text+0x18f8): undefined reference to `specfem::program::abort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, char const*)'
/usr/bin/ld: ../archive/libperiodic_tasks.a(plot_wavefield.cpp.o): in function `void specfem::periodic_tasks::plot_wavefield<(specfem::dimension::type)1>::initialize<(specfem::display::format)3>(vtkSmartPointer<vtkFloatArray>&)':
/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2/core/specfem/periodic_tasks/plot_wavefield/dim3/plot_wavefield.cpp:435:(.text+0x18f8): undefined reference to `specfem::program::abort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, char const*)'
/usr/bin/ld: archive/libperiodic_tasks.a(plot_wavefield.cpp.o): in function `void specfem::periodic_tasks::plot_wavefield<(specfem::dimension::type)1>::initialize<(specfem::display::format)3>(vtkSmartPointer<vtkFloatArray>&)':
/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2/core/specfem/periodic_tasks/plot_wavefield/dim3/plot_wavefield.cpp:435:(.text+0x18f8): undefined reference to `specfem::program::abort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, char const*)'
```
with the following compile flags
```bash
/home/lambda/built_software/cmake/cmake-3.31.7/bin/cmake -S '/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2' -B '/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2/build_debug_serial' -D Kokkos_ENABLE_SERIAL=ON -D CMAKE_BUILD_TYPE=Debug -D SPECFEM_BUILD_TESTS=ON -D CMAKE_C_COMPILER=/home/lambda/built_software/gcc/gcc14.3/usr/local/bin/gcc -D CMAKE_CXX_COMPILER=/home/lambda/built_software/gcc/gcc14.3/usr/local/bin/g++ -D CMAKE_Fortran_COMPILER=/home/lambda/built_software/gcc/gcc14.3/usr/local/bin/gfortran -D CMAKE_EXPORT_COMPILE_COMMANDS=1 -D SPECFEM_ENABLE_UNITY_BUILD=OFF -D SPECFEM_ENABLE_HDF5=ON -D VTK_DIR='/home/lambda/princeton/dg_proj/specfem_dev/SPECFEMPP2/../vtk_940/vtk940_build' -D SPECFEM_ENABLE_VTK=ON
```

--- a/core/specfem/periodic_tasks/CMakeLists.txt
+++ b/core/specfem/periodic_tasks/CMakeLists.txt
@@ -16,6 +16,7 @@ target_link_libraries(
         writer
         io
         assembly
+        specfem::program
 )
 
 target_compile_definitions(

